File: wcserver/c#/examples/BSMiniRPTC#.NET/frmFile.cs
using System;
using System.Collections;
using System.Diagnostics;
using System.Drawing;
using System.Windows.Forms;

using wcSDK;

namespace BSMiniRPT
{
	internal partial class frmFile : System.Windows.Forms.Form
	{


		public short PrepFiles(int lngArea, string strFileName)
		{

			try
			{
				wcServerAPI.TFileRecord wFile = new wcServerAPI.TFileRecord();
				wcServerAPI.TFileArea wFileA = new wcServerAPI.TFileArea();
				int fID = 0;

				if (wcServerAPI.GetFileRecByAreaName(lngArea, strFileName, ref wFile, ref fID))
				{
					if (wcServerAPI.GetFileArea(lngArea, ref wFileA))
					{
						if (wcServerAPI.WcExistFile("wc:\\file\\area(" + wFileA.Number.ToString().Trim() + ")\\" + wFile.Name.Trim()))
						{
							lblFStatus.Text = "**File Exists On Disk**";
						}
						txtFileArea.Text = "(" + wFile.Area.ToString().Trim() + "). " + wFileA.name.Trim();
						txtFileName.Text = wFile.Name.Trim();
						txtFileSize.Text = wFile.Size.ToString("###,###,###,###,###,###,###");
						txtFileUploader.Text = wFile.Uploader.Name.Trim();
						txtFilePassword.Text = wFile.Password.Trim();
						txtFileDate.Text = modBSMiniRPT.DateToDateString(wFile.FileTime, true).Trim();
						txtFileLastAccess.Text = modBSMiniRPT.DateToDateString(wFile.LastAccessed, true).Trim();
						txtFileDownloads.Text = wFile.Downloads.ToString().Trim();
						txtFileDescription.Text = wFile.Description.Trim();
					}
					else
					{
						MessageBox.Show("Error pulling the area description", "WINS Error...", MessageBoxButtons.OK, MessageBoxIcon.Information);
						this.Close();
					}
				}
				else
				{
					MessageBox.Show("Error pulling the selected file", "WINS Error...", MessageBoxButtons.OK, MessageBoxIcon.Information);
					this.Close();
				}
			}
			catch (Exception ex)
			{
				MessageBox.Show("Unexpected Error occurred" + Environment.NewLine + "Error:  " + ex.ToString() + Environment.NewLine + "Message:  " + ex.Message, "Unexpected Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
			}

			return 0;
		}

		private void cmdFiles_Click(object eventSender, System.EventArgs eventArgs)
		{

			this.Dispose();

		}

	}
} //end of root namespace

File: wcserver/vb.net/examples/BSMiniRPT.NET/frmFile.vb
Option Strict On
Option Explicit On

Imports wcSDK.wcServerAPI

Friend Class frmFile

    Inherits System.Windows.Forms.Form

    Public Function PrepFiles(ByVal lngArea As Integer, ByVal strFileName As String) As Short

        Try
            Dim wFile As New TFileRecord
            Dim wFileA As New TFileArea
            Dim fID As Integer

            If GetFileRecByAreaName(lngArea, strFileName, wFile, fID) Then
                If GetFileArea(lngArea, wFileA) Then
                    If WcExistFile("wc:\file\area(" & wFileA.Number.ToString.Trim & ")\" & wFile.name.Trim) Then lblFStatus.Text = "**File Exists On Disk**"
                    txtFileArea.Text = "(" & wFile.area.ToString.Trim & "). " & wFileA.name.Trim
                    txtFileName.Text = wFile.name.Trim
                    txtFileSize.Text = Format(wFile.Size, "###,###,###,###,###,###,###")
                    txtFileUploader.Text = wFile.Uploader.name.Trim
                    txtFilePassword.Text = wFile.Password.Trim
                    txtFileDate.Text = DateToDateString(wFile.FileTime, True).Trim
                    txtFileLastAccess.Text = DateToDateString(wFile.LastAccessed, True).Trim
                    txtFileDownloads.Text = wFile.Downloads.ToString.Trim
                    txtFileDescription.Text = wFile.Description.Trim
                Else
                    MsgBox("Error pulling the area description", MsgBoxStyle.Information, "WINS Error...")
                    Me.Close()
                End If
            Else
                MsgBox("Error pulling the selected file", MsgBoxStyle.Information, "WINS Error...")
                Me.Close()
            End If
        Catch ex As Exception
            MsgBox("Unexpected Error occurred" & Environment.NewLine & "Error:  " & ex.ToString & Environment.NewLine & "Message:  " & ex.Message, MsgBoxStyle.Critical, "Unexpected Error")
        End Try

    End Function

    Private Sub cmdFiles_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdFilesClose.Click

        Me.Dispose()

    End Sub

End Class